internal/fnruntime/runner.go

```diff
@@ -162,6 +172,140 @@ func parseStructuredResult(yml *yaml.RNode, fnResult *fnresult.Result) error {
 	return nil
 }
 
+// printFnResult prints given function result in a user friendly
+// format on kpt CLI.
+func printFnResult(ctx context.Context, fnResult *fnresult.Result) {
+	pr := printer.FromContextOrDie(ctx)
+	printOpt := printer.NewOpt()
+	if len(fnResult.Results) > 0 {
+		// function returned structured results
+		pr.OptPrintf(printOpt, "  Results:\n")
+		var lines []string
+		for _, item := range fnResult.Results {
+			lines = append(lines, resultToString(item))
+		}
+		ri := &multiLineFormatter{
+			Lines:          lines,
+			TruncateOutput: printer.TruncateOutput,
+		}
+		pr.OptPrintf(printOpt, "%s", ri.String())
+	}
+}
+
+// printFnExecErr prints given ExecError in a user friendly format
+// on kpt CLI.
+func printFnExecErr(ctx context.Context, fnErr *ExecError) {
+	pr := printer.FromContextOrDie(ctx)
+	printOpt := printer.NewOpt()
+	if len(fnErr.Stderr) > 0 {
+		pr.OptPrintf(printOpt.Stderr(), "  Stderr:\n")
+		errLines := &multiLineFormatter{
+			Lines:          strings.Split(fnErr.Stderr, "\n"),
+			UseQuote:       true,
+			TruncateOutput: printer.TruncateOutput,
+		}
+		pr.OptPrintf(printOpt.Stderr(), "%s", errLines.String())
+	}
+	pr.OptPrintf(printOpt.Stderr(), "  Exit code: %d\n\n", fnErr.ExitCode)
+}
+
+// multiLineFormatter knows how to format multiple lines in pretty format
+// that can be displayed to an end user.
+type multiLineFormatter struct {
+	// Lines to be printed on the CLI.
+	Lines []string
+
+	// TruncateOuput determines if output needs to be truncated or not.
+	TruncateOutput bool
+
+	// MaxLines to be printed if truncation is enabled.
+	MaxLines int
+
+	// UseQuote determines if line needs to be quoted or not
+	UseQuote bool
+}
+
+// String returns multiline string.
+func (ri *multiLineFormatter) String() string {
+	ri.MaxLines = FnExecErrorTruncateLines
```

Comment:
We can create a function for truncation to avoid duplicates with https://github.com/GoogleContainerTools/kpt/blob/next/internal/errors/fnerrors.go#L51

Reply:
Now `ExecError` uses multiLineFormatter, so no duplication. Thanks.